### ISSUE TITLE
Unable to communicate with Python API

### ISSUE THREAD

**Issue by theoturner** [cypress-io/github-action#330]
## TLDR: Front-end via Cypress Github action is unable to communicate with a Python API, but works fine locally

I run an external API using uvicorn (fastapi/flask) which my front-end reads from. I have set up a test fixture where this loads a small test DB from checked in CSVs. **If I boot this on my local machine, and then run cypress (both interactive and CLI), everything works fine.** In the Github Action, the front-end is unable to send requests to the API. Here is my workflow file:

 ## Workflow file
 
```yaml
name: Front-end and end-to-end tests

on:
  pull_request:
    branches:
      - master

jobs:
  cypress-run:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - uses: actions/setup-python@v2
        with:
          python-version: '3.8'
      - run: pip install -r requirements.txt
        name: Install requirements
      - run: uvicorn --port 8001 src.api.core.main:app &
        name: Run test server
        env:
          PYTHONPATH: ./src/
      - name: Cypress run
        uses: cypress-io/github-action@v2
        with:
          start: npm start
          working-directory: client
      - uses: actions/upload-artifact@v1
         if: always()
         with:
           name: cypress-videos
           path: client/cypress/videos
```

## Cypress run output

```

    1) Check empty player becoming actual player works
    2) Check all position inputs
    3) Check all in position score inputs
    4) Check all minutes inputs
    5) Check all formation inputs
    6) Check all game-level feature inputs


  0 passing (58s)
  6 failing

  1) Integration tests
       Check empty player becoming actual player works:
     AssertionError: Timed out retrying after 4000ms: Expected to find element: `.PitchContainer`, but never found it.
      at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/integration/apiIntegration.js:309:31)

  2) Integration tests
       Check all position inputs:
     AssertionError: Timed out retrying after 4000ms: Expected to find element: `.PitchContainer`, but never found it.
      at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/integration/apiIntegration.js:338:31)

  3) Integration tests
       Check all in position score inputs:
     AssertionError: Timed out retrying after 4000ms: Expected to find element: `.PitchContainer`, but never found it.
      at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/integration/apiIntegration.js:362:31)

  4) Integration tests
       Check all minutes inputs:
     AssertionError: Timed out retrying after 4000ms: Expected to find element: `.PitchContainer`, but never found it.
      at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/integration/apiIntegration.js:383:31)

  5) Integration tests
       Check all formation inputs:
     AssertionError: Timed out retrying after 4000ms: Expected to find element: `.PitchContainer`, but never found it.
      at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/integration/apiIntegration.js:414:31)

  6) Integration tests
       Check all game-level feature inputs:
     AssertionError: Timed out retrying after 4000ms: Expected to find element: `.PitchContainer`, but never found it.
      at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/integration/apiIntegration.js:432:31)




  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        6                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      6                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  6                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     58 seconds                                                                       │
  │ Spec Ran:     apiIntegration.js                                                                │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
```

This same error repeats for all of my test files. **The element with class `PitchContainer` will show up following a successful login, which requires communication with the API.** I know that the credentials entered are correct because cypress runs fine locally, and the video uploaded to GH run artifacts shows that the communication is not working.

## What I've tried

1. Running my API start command under Cypress run's `with: start:` (as with `npm start` above)
2. Running cypress manually via `npx cypress run`
3. Long `cy.wait()` following each button click which makes a request to the API
4. Running on various versions of Ubuntu (`using:` in the workflow file)

In all cases the error is the same. As I understand, the Cypress github action *does* work with an external API, [PostHog uses this config in production by booting up their API in a container](https://github.com/PostHog/posthog/blob/master/.github/workflows/e2e.yml). So, I assume there's a compatibility issue here with the fastapi server.

**Comment by theoturner** [comment]
Problem solved! Two core things did it:
1. The API server was missing a pickle file that I had omitted. As these logs couldn't be captured in Cypress it took a while to find, so for anyone with the same issue make sure your API has all its deps. Fixing this solved some of my failing tests.
2. Playing around with cy.wait() timing - the initial page load was just showing blank on occasion and doing a cy.get() of the page twice and adding a wait after anything that loaded a page or made a request to the server allowed some further tests to pass.
